commit 64d8639925fff0d9d67b068cf80c16f14c0a0ab9 (refs/changes/82/1082/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-12-07T22:08:37+00:00 (2 years, 10 months ago)
    
    joyent/node-lockfd#6 move flock change to 2.x stream, create v1.x branch for 1.x releases (setup 1.x branch)

diff --git a/.gitignore b/.gitignore
index 5a891b0..255673d 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,6 +1,5 @@
 .*.swp
 /0-npm-stamp
-/Makefile
 /downloads/
 /lib/*.node
 /main.js
diff --git a/CHANGES.md b/CHANGES.md
new file mode 100644
index 0000000..951655e
--- /dev/null
+++ b/CHANGES.md
@@ -0,0 +1,31 @@
+# node-lockfd changelog
+
+## not yet released
+
+- joyent/node-lockfd#6: *Removes* the flock(3C) support added in
+  joyent/node-lockfd#5. See the master branch and 2.x releases for
+  that functionality.
+
+## 1.1.0
+
+*WARNING*: This release has been re-branded "2.0.0" because it broke
+compatibility by raising the mininum SmartOS platform requirement to one
+that includes flock(3C) -- OS-2868, platform versions after 20150218.
+
+- joyent/node-lockfd#5 Add support for flock(3C) 
+
+## 1.0.1
+
+- Allow `require('lockfd')` import to work.
+
+## 1.0.0
+
+- Bump v8plus dep to 0.3.1.
+
+## 0.0.3
+
+- Improved error handling.
+
+## earlier
+
+See the commit log.
diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..eea59d1
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,65 @@
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+#
+# Copyright 2016, Joyent, Inc.
+#
+
+#
+# node-lockfd Makefile
+#
+
+#
+# Tools
+#
+
+NPM	:= npm
+
+#
+# Files
+#
+
+LOCKFD_BINDING	:= ./lib/lockfd_binding.node
+
+
+#
+# Repo-specific targets
+#
+
+.PHONY: all
+all: $(LOCKFD_BINDING) ./node_modules
+	$(NPM) install
+
+$(LOCKFD_BINDING):
+	cd src && make
+
+CLEAN_FILES += $(LOCKFD_BINDING)
+
+# Ensure CHANGES.md and package.json have the same version.
+.PHONY: versioncheck
+versioncheck:
+	@echo version is: $(shell cat package.json | json version)
+	[[ `cat package.json | json version` == `grep '^## ' CHANGES.md | head -2 | tail -1 | awk '{print $$2}'` ]]
+
+.PHONY: cutarelease
+cutarelease: versioncheck
+	[[ -z `git status --short` ]]  # If this fails, the working dir is dirty.
+	@which json 2>/dev/null 1>/dev/null && \
+	    ver=$(shell json -f package.json version) && \
+	    name=$(shell json -f package.json name) && \
+	    publishedVer=$(shell npm view -j $(shell json -f package.json name)@$(shell json -f package.json version) version 2>/dev/null) && \
+	    if [[ -n "$$publishedVer" ]]; then \
+		echo "error: $$name@$$ver is already published to npm"; \
+		exit 1; \
+	    fi && \
+	    echo "** Are you sure you want to tag and publish $$name@$$ver to npm?" && \
+	    echo "** Enter to continue, Ctrl+C to abort." && \
+	    read
+	ver=$(shell cat package.json | json version) && \
+	    date=$(shell date -u "+%Y-%m-%d") && \
+	    git tag -a "v$$ver" -m "version $$ver ($$date)" && \
+	    git push --tags origin && \
+	    npm publish
diff --git a/README.md b/README.md
index 231e2da..4c07c86 100644
--- a/README.md
+++ b/README.md
@@ -1,5 +1,10 @@
 # node-lockfd
 
+This repository is part of the Joyent Triton project. See the [contribution
+guidelines](https://github.com/joyent/triton/blob/master/CONTRIBUTING.md) --
+*Triton does not use GitHub PRs* -- and general documentation at the main
+[Triton project](https://github.com/joyent/triton) page.
+
 A trivial wrapper around `fcntl(F_SETLKW)`.  Presently allows
 a synchronous or asynchronous call to get a whole-file, exclusive, advisory
 write lock on a file, or to block until one is possible.
